### PR TITLE
Add a separate process that can handle /sync requests

### PR DESCRIPTION
**Pull request by NegativeMjark**

### CODE REVIEW

File: synapse/app/synchrotron.py [hunk 1]
```diff
@@ -0,0 +1,467 @@
...
+    SlavedApplicationServiceStore,
+    SlavedRegistrationStore,
+    SlavedFilteringStore,
+    SlavedPresenceStore,
+):
+    def get_presence_list_accepted(self, user_localpart):
+        return ()
+
+    def insert_client_ip(self, user, access_token, ip, user_agent):
+        pass
```
Comment (erikjohnston) [inline synapse/app/synchrotron.py]: This might be a useful thing to have

**Comment by erikjohnston** [comment]
It'd probably be nice to poke the master on shutdown, so that presence doesn't get stuck for 5 mins

**Comment by NegativeMjark** [comment]
> It'd probably be nice to poke the master on shutdown, so that presence doesn't get stuck for 5 mins

Fixed 48340e4

@erikjohnston PTAL

**Comment by erikjohnston** [comment]
LGTM

**Pull request merged**